<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1056</width>
    <height>751</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <property name="tabBarAutoHide">
       <bool>false</bool>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Analysis</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayoutDate1">
          <item>
           <widget class="QLabel" name="labelStartDate">
            <property name="text">
             <string>Start Date:</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDateTimeEdit" name="dateTimeEditStart"/>
          </item>
          <item>
           <widget class="QLabel" name="labelEndDate">
            <property name="text">
             <string>End Date:</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDateTimeEdit" name="dateTimeEditEnd"/>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QTabWidget" name="tabWidget_2">
          <property name="currentIndex">
           <number>0</number>
          </property>
          <widget class="QWidget" name="tab_3">
           <attribute name="title">
            <string>Analyze Set Of ID</string>
           </attribute>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_13">
              <item>
               <widget class="QGroupBox" name="groupBoxUser1">
                <property name="title">
                 <string>Show User Range</string>
                </property>
                <layout class="QVBoxLayout" name="verticalLayout_3">
                 <item>
                  <layout class="QHBoxLayout" name="horizontalLayoutUser1">
                   <property name="sizeConstraint">
                    <enum>QLayout::SetDefaultConstraint</enum>
                   </property>
                   <item>
                    <widget class="QLabel" name="label">
                     <property name="text">
                      <string>User ID:</string>
                     </property>
                     <property name="alignment">
                      <set>Qt::AlignCenter</set>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QSpinBox" name="spinBoxUIDSetLow"/>
                   </item>
                   <item>
                    <widget class="QSpinBox" name="spinBoxUIDSetHigh"/>
                   </item>
                  </layout>
                 </item>
                 <item>
                  <widget class="QPushButton" name="btnShowTopPOIFromUser">
                   <property name="text">
                    <string>Show Top 10 POI of user</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
              <item>
               <widget class="QGroupBox" name="groupBoxPOI2">
                <property name="title">
                 <string>Show POI Range</string>
                </property>
                <layout class="QVBoxLayout" name="verticalLayout_12">
                 <item>
                  <layout class="QHBoxLayout" name="horizontalLayoutPOI1_2">
                   <property name="sizeConstraint">
                    <enum>QLayout::SetDefaultConstraint</enum>
                   </property>
                   <item>
                    <widget class="QLabel" name="label_21">
                     <property name="text">
                      <string>POI ID:</string>
                     </property>
                     <property name="alignment">
                      <set>Qt::AlignCenter</set>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QSpinBox" name="spinBoxPOISetLow"/>
                   </item>
                   <item>
                    <widget class="QSpinBox" name="spinBoxPOISetHigh"/>
                   </item>
                  </layout>
                 </item>
                 <item>
                  <widget class="QPushButton" name="btnShowTopUserFromPOI">
                   <property name="text">
                    <string>Show Top 10 user of POI</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QChartView" name="singleChartView"/>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="tab_4">
           <attribute name="title">
            <string>Analyze Users, POI Overtime</string>
           </attribute>
           <layout class="QHBoxLayout" name="horizontalLayout_7">
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_4">
              <item>
               <widget class="QChartView" name="poiSelectView"/>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayoutLatitude1">
                <item>
                 <widget class="QLabel" name="labelLatitude1">
                  <property name="text">
                   <string>Latitude Bound:</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QDoubleSpinBox" name="latitudeBoundXMin">
                  <property name="minimum">
                   <double>-90.000000000000000</double>
                  </property>
                  <property name="maximum">
                   <double>90.000000000000000</double>
                  </property>
                  <property name="value">
                   <double>-90.000000000000000</double>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QDoubleSpinBox" name="latitudeBoundXMax">
                  <property name="minimum">
                   <double>-90.000000000000000</double>
                  </property>
                  <property name="maximum">
                   <double>90.000000000000000</double>
                  </property>
                  <property name="value">
                   <double>90.000000000000000</double>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayoutLongitude1">
                <item>
                 <widget class="QLabel" name="labelLongitude1">
                  <property name="text">
                   <string>Longitude Bound:</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QDoubleSpinBox" name="longitudeBoundXMin">
                  <property name="minimum">
                   <double>-180.000000000000000</double>
                  </property>
                  <property name="maximum">
                   <double>180.000000000000000</double>
                  </property>
                  <property name="value">
                   <double>-180.000000000000000</double>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QDoubleSpinBox" name="longitudeBoundXMax">
                  <property name="minimum">
                   <double>-180.000000000000000</double>
                  </property>
                  <property name="maximum">
                   <double>180.000000000000000</double>
                  </property>
                  <property name="value">
                   <double>180.000000000000000</double>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <widget class="QPushButton" name="btnSetLatLongBound">
                <property name="text">
                 <string>Set Bound</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_5">
              <item>
               <widget class="QChartView" name="totalUserOTView"/>
              </item>
              <item>
               <widget class="QPushButton" name="btnGeneratePOIOvertime">
                <property name="text">
                 <string>Calculate</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="tab_5">
           <attribute name="title">
            <string>Compare Two Set of ID</string>
           </attribute>
           <layout class="QHBoxLayout" name="horizontalLayout_12">
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_8">
              <item>
               <widget class="QGroupBox" name="groupBoxUID1">
                <property name="title">
                 <string>Select User ID</string>
                </property>
                <layout class="QVBoxLayout" name="verticalLayout_6">
                 <item>
                  <layout class="QHBoxLayout" name="horizontalLayout_8">
                   <item>
                    <widget class="QLabel" name="label_7">
                     <property name="text">
                      <string>User 1 ID: </string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QSpinBox" name="spinCompareUID11"/>
                   </item>
                   <item>
                    <widget class="QSpinBox" name="spinCompareUID12"/>
                   </item>
                  </layout>
                 </item>
                 <item>
                  <layout class="QHBoxLayout" name="horizontalLayout_9">
                   <item>
                    <widget class="QLabel" name="label_8">
                     <property name="text">
                      <string>User 2 ID: </string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QSpinBox" name="spinCompareUID21"/>
                   </item>
                   <item>
                    <widget class="QSpinBox" name="spinCompareUID22"/>
                   </item>
                  </layout>
                 </item>
                 <item>
                  <widget class="QPushButton" name="btnCompareUsers">
                   <property name="text">
                    <string>Compare Users</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
              <item>
               <widget class="QGroupBox" name="groupBoxPOI1">
                <property name="title">
                 <string>Select POI ID</string>
                </property>
                <layout class="QVBoxLayout" name="verticalLayout_7">
                 <item>
                  <layout class="QHBoxLayout" name="horizontalLayout_10">
                   <item>
                    <widget class="QLabel" name="label_9">
                     <property name="text">
                      <string>POI 1 ID: </string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QSpinBox" name="spinComparePOI11"/>
                   </item>
                   <item>
                    <widget class="QSpinBox" name="spinComparePOI12"/>
                   </item>
                  </layout>
                 </item>
                 <item>
                  <layout class="QHBoxLayout" name="horizontalLayout_11">
                   <item>
                    <widget class="QLabel" name="label_10">
                     <property name="text">
                      <string>POI 2 ID: </string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QSpinBox" name="spinComparePOI21"/>
                   </item>
                   <item>
                    <widget class="QSpinBox" name="spinComparePOI22"/>
                   </item>
                  </layout>
                 </item>
                 <item>
                  <widget class="QPushButton" name="btnComparePOI">
                   <property name="text">
                    <string>Compare POI</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QChartView" name="compareTwoChartView"/>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
        <item>
         <widget class="QProgressBar" name="progressBar">
          <property name="value">
           <number>0</number>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>Inspection</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_9">
        <item>
         <widget class="QTabWidget" name="tabWidget_3">
          <property name="currentIndex">
           <number>0</number>
          </property>
          <widget class="QWidget" name="tab_6">
           <attribute name="title">
            <string>Trajectory Analysis</string>
           </attribute>
           <layout class="QVBoxLayout" name="verticalLayout_10">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_13">
              <item>
               <widget class="QLabel" name="label_11">
                <property name="text">
                 <string>User ID: </string>
                </property>
                <property name="alignment">
                 <set>Qt::AlignCenter</set>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="spinTrajectoryUID"/>
              </item>
              <item>
               <widget class="QPushButton" name="btnShowTrajectory">
                <property name="text">
                 <string>Show trajectory</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QChartView" name="trajectoryGraphView"/>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_14">
              <item>
               <widget class="QLabel" name="label_12">
                <property name="text">
                 <string>Move:       </string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSlider" name="traSlider">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="maximum">
                 <number>100</number>
                </property>
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_18">
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_16">
                <item>
                 <widget class="QLabel" name="label_14">
                  <property name="text">
                   <string>User Longitude:</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="traLongitude">
                  <property name="readOnly">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_15">
                <item>
                 <widget class="QLabel" name="label_13">
                  <property name="text">
                   <string>User Latitude:</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="traLatitude">
                  <property name="readOnly">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_17">
                <item>
                 <widget class="QLabel" name="label_15">
                  <property name="text">
                   <string>Timestamp:</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="traTimeStamp">
                  <property name="readOnly">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="tab_7">
           <attribute name="title">
            <string>Prediction</string>
           </attribute>
           <layout class="QVBoxLayout" name="verticalLayout_11">
            <item>
             <widget class="QLabel" name="label_16">
              <property name="text">
               <string>Predict total user at a predicted datetime, it'll cache the calculation so you don't have to recalculate the algorithm for a set of ID</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_19">
              <item>
               <widget class="QLabel" name="label_18">
                <property name="text">
                 <string>POI ID Left Bound</string>
                </property>
                <property name="alignment">
                 <set>Qt::AlignCenter</set>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="spinPredictPOI1"/>
              </item>
              <item>
               <widget class="QLabel" name="label_2">
                <property name="text">
                 <string>POI ID Right Bound</string>
                </property>
                <property name="alignment">
                 <set>Qt::AlignCenter</set>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="spinPredictPOI2"/>
              </item>
              <item>
               <widget class="QPushButton" name="btnGenerateCubic">
                <property name="text">
                 <string>Generate Interpolation Cache</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_20">
              <item>
               <widget class="QLabel" name="label_17">
                <property name="text">
                 <string>Currently cached POI ID Range: </string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="labelNewtonCache">
                <property name="text">
                 <string>None</string>
                </property>
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="label_19">
                <property name="text">
                 <string>Predict DateTime: </string>
                </property>
                <property name="alignment">
                 <set>Qt::AlignCenter</set>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QDateTimeEdit" name="predictDate">
                <property name="date">
                 <date>
                  <year>2009</year>
                  <month>9</month>
                  <day>1</day>
                 </date>
                </property>
                <property name="displayFormat">
                 <string>dd/MM/yyyy</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="btnShowPrediction">
                <property name="text">
                 <string>Predict</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QChartView" name="predictionGraphView"/>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_21">
              <item>
               <widget class="QLabel" name="label_20">
                <property name="text">
                 <string>Result: </string>
                </property>
                <property name="alignment">
                 <set>Qt::AlignCenter</set>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="predictResultLabel">
                <property name="text">
                 <string>No message</string>
                </property>
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
        <item>
         <widget class="QProgressBar" name="progressBar_2">
          <property name="value">
           <number>0</number>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QChartView</class>
   <extends>QGraphicsView</extends>
   <header>QtCharts</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
